src/stores/talkdashboard.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { defineStore } from 'pinia'
+
+import { hasTalkFeature } from '../services/CapabilitiesManager.ts'
+import { getDashboardEventRooms } from '../services/dashboardService.ts'
+import { getUpcomingReminders } from '../services/remindersService.js'
+import type { DashboardEventRoom, UpcomingReminder } from '../types/index.ts'
+
+const supportsUpcomingReminders = hasTalkFeature('local', 'upcoming-reminders')
+
+type State = {
+	eventrooms: DashboardEventRoom[],
+	upcomingReminders: UpcomingReminder[],
+}
+export const useTalkDashboardStore = defineStore('talkdashboard', {
+	state: (): State => ({
+		eventrooms: [],
+		upcomingReminders: [],
+	}),
+
+	getters: {
+		getEventRooms: (state) => state.eventrooms,
+		getUpcomingReminders: (state) => state.upcomingReminders,
+	},
+
+	actions: {
+		async fetchDashboardEventRooms() {
+			try {
+				const response = await getDashboardEventRooms()
+				this.eventrooms = response.data.ocs.data
+				return this.eventrooms
```

Comment:
We can return from the store action, but in that case I would rely on getter and do interval requests to update the state
```suggestion
				Vue.set(this, 'eventRooms', response.data.ocs.data)
```